app/Http/Controllers/Assets/BulkAssetsController.php

```diff
@@ -208,61 +208,86 @@ public function update(Request $request)
                     }
                 }
-
-                $logAction = new Actionlog();
-                $logAction->item_type = Asset::class;
-                $logAction->item_id = $assetId;
-                $logAction->created_at =  date("Y-m-d H:i:s");
-                $logAction->user_id = Auth::id();
-                $logAction->log_meta = json_encode($changed);
-                $logAction->logaction('update');
  
-                if($custom_fields_present) { 
-                    $asset = Asset::find($assetId); 
-                    $assetCustomFields = $asset->model()->first()->fieldset; 
-                    if($assetCustomFields && $assetCustomFields->fields) {
-                            foreach ($assetCustomFields->fields as $field) { 
-                                if (array_key_exists($field->db_column, $this->update_array)) {
-                                        $asset->{$field->db_column} = $this->update_array[$field->db_column]; 
-                                        $saved = $asset->save();    
-                                        if(!$saved) {
-                                            $error_bag[] = $asset->getErrors();
+                if ($custom_fields_present) {
+
+                    $model = $asset->model()->first();
+
+                    // Use the rules of the new model fieldsets if the model changed
+                    if ($request->filled('model_id')) {
+                        $this->update_array['model_id'] =  $request->input('model_id');
+                        $model = \App\Models\AssetModel::find($request->input('model_id'));
+                    }
+
+
+                    // Make sure this model is valid
+                    $assetCustomFields = ($model) ? $model->fieldset : null;
+
+                    if ($assetCustomFields && $assetCustomFields->fields) {
+
+                            foreach ($assetCustomFields->fields as $field) {
+
+                                if ((array_key_exists($field->db_column, $this->update_array)) && ($field->field_encrypted=='1')) {
+                                    $decrypted_old = Helper::gracefulDecrypt($field, $asset->{$field->db_column});
+
+                                    // Check if the decrypted existing value is different from one we just submitted
+                                    // and if not, pull it out of the object
+                                    if ($decrypted_old != $this->update_array[$field->db_column]) {
+                                        $asset->{$field->db_column} = \Crypt::encrypt($this->update_array[$field->db_column]);
+                                    } else {
+                                        \Log::debug('The decrypted existing value is the same as the one submitted - unset it');
+                                        unset($this->update_array[$field->db_column]);
+                                        unset($asset->{$field->db_column});
```

Comment:
This spooked me for a second and made me think that we were *blanking* the custom field property - but we're not, we're saying it's not-changed and thus doesn't need to get re-encrypted here. I think a comment for something like that would be helpful.